opencog/nlp/relex2logic/det.scm

```diff
@@ -0,0 +1,59 @@
+
+(define det
+    (BindLink
+        (VariableList
+            (TypedVariableLink
+                (VariableNode "$a-parse")
+                (TypeNode "ParseNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$noun")
+                (TypeNode "WordInstanceNode")
+            )
+            (TypedVariableLink
+                (VariableNode "$det")
+                (TypeNode "WordInstanceNode")
+            )
+        )
+        (AndLink
+            (WordInstanceLink
+                (VariableNode "$noun")
+                (VariableNode "$a-parse")
+            )
+            (WordInstanceLink
+                (VariableNode "$det")
+                (VariableNode "$a-parse")
+            )
+            (EvaluationLink
+                (DefinedLinguisticRelationshipNode "_det")
+                (ListLink
+                    (VariableNode "$noun")
+                    (VariableNode "$det")
+                )
+            )
+            (InheritanceLink 
+                (VariableNode "$noun")
```

Comment:
@misgeatgit @leungmanhin @AmeBel I am aware that for R2L we have agreed to apply all rules in one step, but I don't recall deciding to do this only once.  Post-processing was always in the pipeline and I always assume there will be more than one step.  ie.  all rules applied at step 1, then again all rules at step 2 (and maybe again at step 3 or 4 depends on how many level of post-processing is needed).  Maybe @bgoertzel will have some input for this?

Of cause it is possible to do `cog-fc` multiple times only different rule bases externally but that seems to be wasting the functionality of the chainer.  Maybe there some performance benefit for this, but it seems very minor.